lib/iris/plot.py

```diff
@@ -403,8 +405,14 @@ def _draw_2d_from_points(draw_method_name, arg_func, cube, *args, **kwargs):
 
 def _fixup_dates(coord, values):
     if coord.units.calendar is not None and values.ndim == 1:
-        r = [datetime.datetime(*(coord.units.num2date(val).timetuple()[0:6]))
-             for val in values]
+        num2date = coord.units.num2date
+        if coord.units.calendar == 'gregorian':
```

Comment:
Right now, nc_time_axis doesn't provide all the functionality that matplotlib can do with datetime.datetimes (e.g. extended handling of tick locators). Until this is the case, it would be best to make use of datetimes for gregorian calendars.